sbom-utility-scripts/scripts/index-image-sbom-script/index_image_sbom_script.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+import argparse
+import json
+from datetime import datetime, timezone
+from typing import Optional
+from dataclasses import dataclass, field
+from uuid import uuid4
+
+from packageurl import PackageURL
+
+
+SUPPORTED_ARCHITECTURES = ["amd64", "arm64", "s390x", "ppc64le"]
```

Comment:
This script should be architecture agnostic. Ideally, there should not be any hardcoded value for an architecture. The pipeline itself defines which architectures will be built so this script should just take the output from the pipeline task and assemble the sbom. In case a new arch is added or anything related to the arch is modified this script becomes a blocker for the rest of the pipeline.

Reply:
Thanks for pointing that out. I have reworked the script so now it works even if no architecture of the child image is specified and no value is hardcoded. It just produces one less PURL if that's the case. If you have further points to improve this, be sure to let me know :saluting_face:

Reply:
I have reworked it again, now it should work with the output of command `buildah manifest inspect` instead of the long list of arguments as it was previously.